framework_tvb/tvb/core/services/project_service.py

```diff
@@ -851,6 +852,33 @@ def get_operations_for_datatype(datatype_gid, visibility_filter, only_in_groups=
             return dao.get_operations_for_datatype(datatype_gid, only_relevant=False, only_in_groups=only_in_groups)
         return dao.get_operations_for_datatype(datatype_gid, only_in_groups=only_in_groups)
 
+    def cancel_or_remove_burst(self, burst_id):
+        """
+        Cancel (if burst is still running) or Remove the burst given by burst_id.
+        :returns True when Remove operation was done and False when Cancel
+        """
+        burst_entity = dao.get_burst_by_id(burst_id)
+        if burst_entity.status == burst_entity.BURST_RUNNING:
+            self.stop_burst(burst_entity.fk_simulation)
+            return False
+
+        ## Remove each DataType in current burst.
+        datatypes = dao.get_results_for_operation(burst_entity.fk_simulation)
+
+        # Remove burst first to delete work-flow steps which still hold foreign keys to operations.
+        correct = dao.remove_entity(burst_entity.__class__, burst_id)
+        if not correct:
+            raise RemoveDataTypeException("Could not remove Burst entity!")
+
+        for datatype in datatypes:
+            ProjectService().remove_datatype(burst_entity.fk_project, datatype.gid, False)
+
+    def stop_burst(self, operation_id):
+        """
+        Stop the operation given by the operation id.
+        """
+        return BACKEND_CLIENT.stop_operation(int(operation_id))
```

Comment:
Call stop_operation here.
Also, shouldn't this handle stop also for PSE bursts?

Reply:
The implemented code works for stop/cancel and also remove for both normal and PSE bursts.